pandas-stubs/core/groupby/generic.pyi

```diff
@@ -46,6 +48,12 @@ def pin_whitelisted_properties(
     klass: Type[FrameOrSeries], whitelist: FrozenSet[str]
 ): ...
 
+class SeriesGroupByScalar(SeriesGroupBy):
+    def __iter__(self) -> Iterator[Tuple[Scalar, Series]]: ...
+
+class SeriesGroupByNonScalar(SeriesGroupBy):
+    def __iter__(self) -> Iterator[Tuple[Tuple, Series]]: ...
```

Comment:
No strong opinion: I see the point from a user-perspective, but I think deviating too much from the pandas implementation will make it challenging to use stubtest later (and maybe also maintaining the stubs).

Reply:
should probably make these classes that are only in the stubs but not in pandas private

Reply:
> No strong opinion

I think I develop a slightly stronger opinion. I would prefer not to introduce too many non-pandas classes: I believe that this will keep pandas-stubs more maintainable in the long-run.

Personally, I prefer the previous state of the PR where you returned `tuple[Hashable, NDFrameT]` (or making SeriesGroupBy generic, might be more invasive).

Reply:
> should probably make these classes that are only in the stubs but not in pandas private

Did that in the next commit

Reply:
While I prefer pandas and pandas-stubs to be aligned (I would like if pandas and pandas-stub could in the future converge and only be different in challenging cases: `__getitem__, Timestamp, ...` - might be a long list) I'm fine with introducing classes/variables that are not in pandas as long as they are private (or some other prefix).